src/libraries/Microsoft.PowerFx.Core/Lexer/TexlLexer.cs

```diff
@@ -76,316 +68,67 @@ public enum Flags
-            AddKeyword(KeywordSelf, TokKind.Self);
-            AddKeyword(KeywordParent, TokKind.Parent);
-            AddKeyword(KeywordAnd, TokKind.KeyAnd);
-            AddKeyword(KeywordOr, TokKind.KeyOr);
-            AddKeyword(KeywordNot, TokKind.KeyNot);
-            AddKeyword(KeywordAs, TokKind.As);
-
-            PopulateKeywordArrays();
-        }
+        private static readonly Lazy<TexlLexer> _invariantDecimalSeparatorLexer = new Lazy<TexlLexer>(() => new TexlLexer(PunctuatorDecimalSeparatorInvariant));
+        private static readonly Lazy<TexlLexer> _commaDecimalSeparatorLexer = new Lazy<TexlLexer>(() => new TexlLexer(PunctuatorCommaInvariant));
 
-        // This creates a new Lexer with the same locale as the original, which is a lot less expensive than spinning up a whole new one
-        private TexlLexer(TexlLexer original)
-        {
-            Contracts.AssertValue(original);
+        public static TexlLexer InvariantLexer => _invariantDecimalSeparatorLexer.Value;
 
-            LocalizedPunctuatorDecimalSeparator = original.LocalizedPunctuatorDecimalSeparator;
-            LocalizedPunctuatorListSeparator = original.LocalizedPunctuatorListSeparator;
-            LocalizedPunctuatorChainingSeparator = original.LocalizedPunctuatorChainingSeparator;
+        private static readonly IReadOnlyList<string> _unaryOperatorKeywords;
+        private static readonly IReadOnlyList<string> _binaryOperatorKeywords;
+        private static readonly IReadOnlyList<string> _operatorKeywordsPrimitive;
+        private static readonly IReadOnlyList<string> _operatorKeywordsAggregate;
+        private static readonly IReadOnlyList<string> _constantKeywordsDefault;
+        private static readonly IReadOnlyList<string> _constantKeywordsGetParent;
 
-            _thousandSeparator = original._thousandSeparator;
+        private readonly IReadOnlyDictionary<string, TokKind> _punctuators;
+        private readonly char _decimalSeparator;
+        private readonly IReadOnlyDictionary<string, string> _punctuatorsAndInvariants;
+        private readonly NumberFormatInfo _numberFormatInfo;
 
-            _cache = null;
+        public string LocalizedPunctuatorDecimalSeparator { get; }
 
-            // Note that the following objects are NOT Cloned, they are the same as the originals, as they are never written to after initialization
-            _keywords = original._keywords;
-            _punctuators = original._punctuators;
-            _decimalSeparator = original._decimalSeparator;
-            Culture = original.Culture;
+        public string LocalizedPunctuatorListSeparator { get; }
 
-            PopulateKeywordArrays();
-        }
+        public string LocalizedPunctuatorChainingSeparator { get; }
 
-        private void AddKeyword(string str, TokKind tid)
-        {
-            Contracts.AssertNonEmpty(str);
-            _keywords.Add(str, tid);
-        }
+        private Tuple<string, Flags, Token[]> _cache;
```

Comment:
This Cache is the last thing I need to clean up to mark TexlLexer  as ThreadSafeImmutable

Reply:
Do we even need the _cache here? it only handles a single item- how often do we actually get a cache hit?

Reply:
Yeah, agreed. Removed the cache, seems like it's not a huge benefit to maintain.